Newsletters: add 357 (2025-06-06) (#2353)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-06-06-newsletter.md

  nodes with a particular configuration to avoid downloading some
  historic blockchain data.  By default, Bitcoin Core nodes use the
  `assumevalid` configuration setting that skips validation of scripts
  in blocks created more than a month or two before the release of the
  version of Bitcoin Core being run.  Although disabled by default, many
  users of Bitcoin Core also set a `prune` configuration setting that
  deletes blocks some time after validating them (how long blocks are
  kept depends on the size of the blocks and specific setting selected
  by the user).

  SK argues that witness data, which is only used for validating
  scripts, should not be downloaded by pruned nodes for assumevalid
  blocks because they won't use it for validating scripts and will
  eventually delete it.  Skipping download of witnesses "can cut
  bandwidth usage by over 40%," he writes.

  Ruben Somsen [argues][somsen nowit] that this does change the security
  model, at least some.  Although scripts aren't validated, the
  downloaded data is validated against the commitment from the block
  header merkle root to the coinbase transaction to the witness data.

>   This ensures the data was available and uncorrupted at the time the

Author: Leaving as-is.  "Were available" feels wrong to me, but I don't know the rule either way.

  The discussion was ongoing at the time of writing.

## Changing consensus

_A monthly section summarizing proposals and discussion about changing
Bitcoin's consensus rules._

- **Quantum computing report:** Clara Shikhelman [posted][shikelman
  quantum] to Delving Bitcoin the summary of a [report][sm report] she
  co-authored with Anthony Milton about the risks to Bitcoin user of
  fast quantum computers, an overview of several pathways to [quantum
  resistance][topic quantum resistance], and an analysis of tradeoffs
  involved in upgrading the Bitcoin protocol.  The authors find 4 to 10
  million BTC are potentially vulnerable to quantum theft, some
  mitigation now is possible, Bitcoin mining is unlikely to be
  threatened by quantum computing in the short or medium term, and
  upgrading requires widespread agreement.

- **Transaction weight limit with exception to prevent confiscation:**
  Vojtěch Strnad [posted][strnad limit] to Delving Bitcoin to propose
  the idea for a consensus change to limit the maximum weight of most
  transactions in a block.  The simple rule would only allow one
  transaction per block (besides the coinbase transaction) to exceed

>   400,000 weight units (100,000 vbytes).  Strnad and others described

Reviewer: This reads like there can be more transactions in the block along with the oversized one. Crucially, the rule would forbid such a block from having any other transactions.

Author: I didn't realize that, sorry!  Will revise.

  the motivation for limiting the maximum transaction weight:

  - _Easier block template optimization:_ it's easier to find a
    near-optimal solution to the [knapsack problem][] the smaller the
    items are compared to the size of the overall limit.  This is partly
    due to minimizing the amount of space leftover at the end, with
    smaller items leaving less space leftover.

  - _Easier relay policy:_ the policy for relaying unconfirmed
    transactions between nodes tries to guess what transactions will be
    mined in order to avoid wasting bandwidth.  Giant transactions make
    guessing harder as even a small change in the top feerate can cause
    them to be delayed or evicted.

  Gregory Sanders [noted][sanders limit] it might be reasonable to
  simply soft fork a maximum weight limit without any exceptions based
  on Bitcoin Core's 12 years of consistent relay policy.  Gregory
  Maxwell [added][maxwell limit] that transactions spending only UTXOs
  created before the soft fork could be allowed an exception to prevent
  confiscation, and that a [transitory soft fork][topic transitory soft

>   forks] could be used to allow the restriction to expire if the

Reviewer 2: ```suggestion
  forks] would allow the restriction to expire if the
```

  community decided not to renew it.

>   Additional discussion examined the needs of the parties who might want

Reviewer 2: ```suggestion
  Additional discussion examined the needs of parties wanting
```

>   large transactions, mainly [BitVM][topic acc] users in the near-term,

Reviewer 2: ```suggestion
  large transactions, mainly [BitVM][topic acc] users in the near term,
```
(Noun versus adjective; I think "near-term" would be correct in a sentence like, "Near-term considerations lead us to conclude...")

  and whether alternative approaches were available to them.

- **Removing outputs from the UTXO set based on value and time:** Robin
  Linus [posted][linus dust] to Delving Bitcoin to propose a soft fork

>   for removing low-value outputs from the UTXO set after a period of

Reviewer 2: ```suggestion
  for removing low-value outputs from the UTXO set after some
```

  time.  Several variations on the idea were discussed, with the two
  main alternatives being:

  - _Destroy old uneconomic funds:_ small value outputs that had not

>     been spent for a period of time would become unspendable.

Reviewer 2: ```suggestion
    been spent for a long time would become unspendable.
```

  - _Require old uneconomic funds to be spent with a proof of existence:_
    [utreexo][topic utreexo] or a similar system could be used to allow
    a transaction to prove that the outputs it spends are part of the
    UTXO set.  Old and [uneconomical][topic uneconomical outputs] would
    need to include this proof, but newer and higher-value outputs would
    still be stored in the UTXO set.

  Either solution would effectively limit the maximum size of the UTXO

>   set (assuming a minimum value and the limit of 21 million bitcoins).

Reviewer 2: ```suggestion
  set (assuming a minimum value and the 21 million bitcoins limit).
```
Or "bitcoin limit"? Not sure.
